dashboard/app/models/user.rb

```diff
@@ -816,6 +816,14 @@ def oauth_only?
     end
   end
 
+  def google_oauth_email_matches?
+    if migrated?
+      authentication_options.any?(&:google_oauth?) && authentication_options.find(&:google_oauth?).email == email
```

Comment:
I think this is both more terse and more flexible in the case that an account has multiple linked Google accounts (a possibility in our data model, although our UI doesn't encourage it.)
```suggestion
      authentication_options.any? {|ao| ao.google_oauth? && ao.email == email}
```

Reply:
Oh yeah that's way better!